src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlCommand.cs

```diff
@@ -3084,6 +3084,12 @@ internal void DeriveParameters()
                         p.TypeName = r[colNames[(int)ProcParamsColIndex.TypeCatalogName]] + "." +
                             r[colNames[(int)ProcParamsColIndex.TypeSchemaName]] + "." +
                             r[colNames[(int)ProcParamsColIndex.TypeName]];
+
+                        // the constructed type name above is incorrectly formatted, it should be a 2 part name not 3
+                        // for compatibility we can't change this because the bug has existed for a long time and been 
```

Comment:
In general, I believe we have to address the bug inside the driver instead of bringing the work around inside.

Reply:
If you look up the original fix in corefx you'll find that there was a lot of conversation around it. 

the reason it's fixed this late is that it's a long standing bug and if users have worked around it by assigning the TypeName directly themselves then we'd break their code. So what we do it follow the original pattern and leave it broken in case users are going to fix it and then at the very last chance before we're about to send the query to the server we check if they've fixed it themselves and if they haven't we fix it for them. It'd be simpler to fix it directly but we'd annoy a lot of people, this way everyone stays working and no-one new has the problem.

Reply:
I am still hesitant to approve the PR although it looks good to me logic wise. for below reasons:

- It may cause some problems for unseen cases in future.
- It will need more debugging and understanding of this workaround logic, which we do not need to deal with if we fix the issue.
- We are fixing things for client without them knowing.

Will it be too much if ask you to make a draft/regular PR with the actual fix then we can decide on how we are going to proceed?
I will ask @David-Engel opinion on this as well.

Just to make this easy to follow, the issue #1001 was opened in Microsoft.Data.SqlClient Github and that was referring to a [problem in corefx](https://github.com/dotnet/corefx/pull/35549)

**System.Data.SqlClient has accepted the workaround.** and apparently the sample repro provided in issue #1001  works fine using that library.

Reply:
The unconditional fix version would simply be to change the type name derivation to:
```csharp
p.TypeName = r[colNames[(int)ProcParamsColIndex.TypeSchemaName]] + "." +
                            r[colNames[(int)ProcParamsColIndex.TypeName]];
```
and add the covering test.

You're right that this method of fixing it was chosen with compatibility as a major factor. We have the chance now to make a breaking change. I don't have a strong opinion on that. Lets see what David thinks.

Reply:
Reading through the old discussion on corefx about the behavior of the implemented fix and in-use workarounds, I'm good with approving this PR as-is.